clash-lib/src/Clash/Netlist.hs

```diff
@@ -769,10 +807,11 @@ mkDcApplication
     -- ^ Returned expression and a list of generate BlackBox declarations
 mkDcApplication dstHType bndr dc args = do
   let dcNm = nameOcc (dcName dc)
-  tcm                 <- Lens.use tcCache
-  let argTys          = map (termType tcm) args
+  tcm <- Lens.use tcCache
+  let argTys = map (termType tcm) args
   argNm <- either return (\b -> extendIdentifier Extended (nameOcc (varName b)) "_dc_arg") bndr
-  argHWTys            <- mapM coreTypeToHWTypeM' argTys
+  argHWTys <- mapM coreTypeToHWTypeM' argTys
+  voidDecls <- renderVoids (zip argHWTys args)
```

Comment:
Same, can't we filter later so that these `voidDecls` are included in `argDecls`?